url.bs

```diff
@@ -197,9 +201,17 @@ U+005B ([) to U+005E (^), inclusive, and U+007C (|).
  concatenated, in the same order.
 </ol>
 
-<p class="note no-backref">The <a><code>application/x-www-form-urlencoded</code></a> format's
-<a lt="urlencoded byte serializer">byte serializer</a> and the <a>URL parser</a>'s
-<a>query state</a> use <a>percent encode</a> directly.
+<p>To <dfn>string UTF-8 percent encode</dfn> a <a for=/>string</a> <var>input</var> using a
```

Comment:
IMO this should also be named "UTF-8 percent encode", distinguishing between the two with `for="code point"` vs. `for=string`. I.e., you should overload the algorithm on its input type.

This is especially intuitive here because the reader then blurs the difference between code points and single-character strings, which is a reasonable mental model that matches many programming languages.

Reply:
You could even rename "string percent decode" to just "percent decode" with for=string. And then you could add a note afterward saying something like

> In general, percent encoding results in a string, whereas percent decoding results in a byte sequence.

Reply:
Maybe even

> In general, percent encoding results in a string with "more percent signs" than the input byte/code point/string, and percent decoding results in a byte sequence with "less percent signs" than the input string/byte sequence.